core/src/main/java/com/linecorp/armeria/client/Clients.java

```diff
@@ -182,6 +182,22 @@
      * @see ClientBuilder ClientBuilder, for more information about how the base options and
      *                    additional options are merged when a derived client is created.
      */
+    public static <T> T newDerivedClient(T client,
+                                         ClientBuilderParams currentParams,
```

Comment:
I went with trying a pattern of adding the ability for ClientFactorys to extract ClientBuilderParams from recognized stubs. Let me know what you think.